keyboards/cassette42/common/oled_helper.c

```diff
@@ -0,0 +1,47 @@
+#ifdef OLED_DRIVER_ENABLE
+#    include QMK_KEYBOARD_H
+#    include <stdio.h>
+#    include <string.h>
+
+void render_logo(void) {
+    static char logo[] = {0x80, 0x81, 0x82, 0x83, 0x84, 0x85, 0x86, 0x87, 0x88, 0x89, 0x8a, 0x8b, 0x8c, 0x8d, 0x8e, 0x8f, 0x90, 0x91, 0x92, 0x93, 0x94, 0xa0, 0xa1, 0xa2, 0xa3, 0xa4, 0xa5, 0xa6, 0xa7, 0xa8, 0xa9, 0xaa, 0xab, 0xac, 0xad, 0xae, 0xaf, 0xb0, 0xb1, 0xb2, 0xb3, 0xb4, 0};
+    oled_write(logo, false);
+}
+
+static char keylog_buf[24]   = "Key state ready.";
+const char code_to_name[60] = {' ', ' ', ' ', ' ', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0', 'R', 'E', 'B', 'T', ' ', '-', ' ', '@', ' ', ' ', ' ', ';', ':', ' ', ',', '.', '/', ' ', ' ', ' '};
+
+void update_key_status(uint16_t keycode, keyrecord_t *record) {
+    if (!record->event.pressed) return;
+    char name = (keycode < 60) ? code_to_name[keycode] : ' ';
+    snprintf(keylog_buf, sizeof(keylog_buf) - 1, "Key:%dx%d %2x %c", record->event.key.row, record->event.key.col, (uint16_t)keycode, name);
+}
+
+void render_key_status(void) { oled_write(keylog_buf, false); }
+
+static char lock_buf[24] = "Lock state ready.\n";
+void update_lock_status(void) {
+    uint8_t leds      = host_keyboard_leds();
+    char *  num_lock  = (leds & (1 << USB_LED_NUM_LOCK)) ? "Num" : "";
+    char *  caps_lock = (leds & (1 << USB_LED_CAPS_LOCK)) ? "Caps" : "";
+    char *  scrl_lock = (leds & (1 << USB_LED_SCROLL_LOCK)) ? "Scrn" : "";
+    snprintf(lock_buf, sizeof(lock_buf) - 1, "Lock:%s %s %s\n", num_lock, caps_lock, scrl_lock);
+}
```

Comment:
This should be converted to something closer to:
```c
    uint8_t led_usb_state = host_keyboard_leds();
    oled_write_P(IS_LED_ON(led_usb_state, USB_LED_NUM_LOCK) ? PSTR("NUMLCK ") : PSTR("       "), false);
    oled_write_P(IS_LED_ON(led_usb_state, USB_LED_CAPS_LOCK) ? PSTR("CAPLCK ") : PSTR("       "), false);
    oled_write_P(IS_LED_ON(led_usb_state, USB_LED_SCROLL_LOCK) ? PSTR("SCRLCK ") : PSTR("       "), false);
```

Reply:
This keypad doesn't use some functions including this, so I'll remove them.
Thanks!

Reply:
Removed
https://github.com/qmk/qmk_firmware/pull/10562/commits/53a4107b3dc36c5d48eaa0cd7cb76cc30efe041a